mods/carts/cart_entity.lua

```diff
@@ -192,11 +196,11 @@ local function rail_on_step(self, dtime)
 	end
 
 	local pos = self.object:get_pos()
-	local cart_dir = carts:velocity_to_dir(vel)
-	local same_dir = vector.equals(cart_dir, self.old_dir)
+	local dir = carts:velocity_to_dir(vel)
+	local dir_changed = not vector.equals(dir, self.old_dir)
```

Comment:
I found `dir_same` more readable. What about `dir_unchanged`?

Reply:
Subjective matter. I'd like to avoid negations within variable names to introduce another lever of confusion. At the end of the day it does not matter at all.